scheduler/scheduler.go

```diff
@@ -151,30 +151,36 @@ func spawnHosts(ctx context.Context, newHostsNeeded map[string]int) (map[string]
 		}
 
 		hostsSpawnedPerDistro[distroId] = make([]host.Host, 0, numHostsToSpawn)
-		// if distro can have containers, check if there are enough parents to hold
-		// new containers
-		if d.MaxContainers > 0 {
-			currentParents, err := host.FindAllRunningParentsByDistro(distroId)
+
+		// if distro is container distro, check if there are enough parent hosts to
+		// support new containers
+		if pool != (evergreen.ContainerPool{}) {
```

Comment:
It's slightly awkward to sometimes pass pools as a pointer and sometimes by value. Here you check `!= (evergreen.ContainerPool{})` but elsewhere you check `!= nil` . I think the latter case is what you want everywhere, unless there's a strong reason to mix.

Reply:
I'll make changes to use pointers when passing around the container pool.